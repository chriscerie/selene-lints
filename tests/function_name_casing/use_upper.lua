local function foo() end
local function foo(bar) end
local function fOO(bAR) end
local function fOo(bAr) end
local function Foo() end
local function Foo(bar) end

function X:a() end
function x:A() end

function X.X:a() end
function x.x:A() end

function X.a() end
function x.A() end

function x.x.a() end
function X.X.A() end

a()
a.a()
A()
A.A()
