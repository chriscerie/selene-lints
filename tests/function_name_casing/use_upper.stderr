error[function_name_casing]: function names must start with an uppercase letter
  ┌─ use_upper.lua:1:16
  │
1 │ local function foo() end
  │                ^^^
  │
  = try: Foo(...) instead

error[function_name_casing]: function names must start with an uppercase letter
  ┌─ use_upper.lua:2:16
  │
2 │ local function foo(bar) end
  │                ^^^
  │
  = try: Foo(...) instead

error[function_name_casing]: function names must start with an uppercase letter
  ┌─ use_upper.lua:3:16
  │
3 │ local function fOO(bAR) end
  │                ^^^
  │
  = try: FOO(...) instead

error[function_name_casing]: function names must start with an uppercase letter
  ┌─ use_upper.lua:4:16
  │
4 │ local function fOo(bAr) end
  │                ^^^
  │
  = try: FOo(...) instead

error[function_name_casing]: function names must start with an uppercase letter
  ┌─ use_upper.lua:8:12
  │
8 │ function X:a() end
  │            ^
  │
  = try: A(...) instead

error[function_name_casing]: function names must start with an uppercase letter
   ┌─ use_upper.lua:11:14
   │
11 │ function X.X:a() end
   │              ^
   │
   = try: A(...) instead

error[function_name_casing]: function names must start with an uppercase letter
   ┌─ use_upper.lua:14:12
   │
14 │ function X.a() end
   │            ^
   │
   = try: A(...) instead

error[function_name_casing]: function names must start with an uppercase letter
   ┌─ use_upper.lua:17:14
   │
17 │ function x.x.a() end
   │              ^
   │
   = try: A(...) instead

